File: distributed-java-rmi/src/main/java/it.unibo.agar.rmi/view/GlobalView.java
package it.unibo.agar.rmi.view;

import it.unibo.agar.rmi.model.GameStateManager;

import javax.swing.*;
import java.awt.*;

public class GlobalView extends JFrame {

    //private final GamePanel gamePanel;

    public GlobalView(GameStateManager gameStateManager) {
        setTitle("Agar.io - Global View (Java)");
        setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE); // Or DISPOSE_ON_CLOSE if multiple windows
        setPreferredSize(new Dimension(800, 800));

        //this.gamePanel = new GamePanel(gameStateManager);
        //add(this.gamePanel, BorderLayout.CENTER);

        pack();
        setLocationRelativeTo(null);
    }

//    public void repaintView() {
//        if (gamePanel != null) {
//            gamePanel.repaint();
//        }
//    }
}

File: distributed-akka/src/main/scala/it/unibo/agar/distributed/view/GlobalView.scala
package it.unibo.agar.distributed.view

import it.unibo.agar.distributed.model.DistributedWorld

import java.awt.Graphics2D
import scala.swing.{Dimension, MainFrame, Panel}

class GlobalView(world: DistributedWorld) extends MainFrame:

  title = "Agar.io - Global View"
  preferredSize = new Dimension(800, 800)

  contents = new Panel:
    override def paintComponent(g: Graphics2D): Unit =
      AgarViewUtils.drawWorld(g, world)
